src/main/java/io/github/cottonmc/cotton/gui/widget/WSprite.java

```diff
@@ -151,4 +151,8 @@ public void paint(MatrixStack matrices, int x, int y, int mouseX, int mouseY) {
 			this.lastFrame = now;
 		}
 	}
+
+	public void paintFrame(int x, int y, Identifier texture) {
```

Comment:
This and the override in `WTiledSprite` should have `@Environment(EnvType.CLIENT)`.